src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java

```diff
@@ -1654,6 +1655,56 @@ public void writeClassDefinition_statementsAndMethods() {
             "}\n"));
   }
 
+  @Test
+  public void writeMethodReturnThisObjectValue() {
+    VaporReference ref =
+        VaporReference.builder().setName("Student").setPakkage("com.google.example.v1").build();
+    TypeNode classType = TypeNode.withReference(ref);
+    MethodDefinition methodDefinition =
+        MethodDefinition.builder()
+            .setName("apply")
+            .setScope(ScopeNode.PUBLIC)
+            .setReturnType(TypeNode.withReference(ref))
+            .setReturnExpr(
+                ValueExpr.builder().setValue(ThisObjectValue.withType(classType)).build())
+            .build();
+    methodDefinition.accept(writerVisitor);
+    assertEquals(
+        writerVisitor.write(),
+        String.format("public Student apply() {\n" + "return this;\n" + "}\n"));
+  }
+
+  @Test
+  public void writeExprStatementForThisObjectValue() {
+    // Example of ThisObjectValue access its fields
```

Comment:
Nit: Usual grammar rules apply to comments, e.g. Example of ThisObjectValue accessing its fields. <-
Same below please.